db/migrate/20201125123131_conv_dpid_seq_for_existing_accnts.rb

```diff
@@ -0,0 +1,46 @@
+class ConvDpidSeqForExistingAccnts < ActiveRecord::Migration[6.0]
+  def up # rubocop:disable Metrics/MethodLength
+    connection.execute("
+    do $$
+    declare
+       disp_id_start integer;
+       max_accnt_id integer;
+       curr_accnt_id integer := 0 ;
+    begin
+        execute 'select MAX(id) from accounts' into max_accnt_id;
+        IF max_accnt_id IS NOT NULL then
+            loop
+                exit when curr_accnt_id > max_accnt_id-1 ;
+                curr_accnt_id := curr_accnt_id + 1 ;
+                execute format('select MAX(display_id) from conversations where account_id=%s', curr_accnt_id) into disp_id_start;
+                IF disp_id_start IS NULL then
+                    disp_id_start := 0;
+                END IF;
+                execute 'create sequence IF NOT EXISTS conv_dpid_seq_' || curr_accnt_id || ' START ' || disp_id_start+1;
+            end loop;
+        END IF;
+        raise notice '%', curr_accnt_id;
+    end; $$;
+    ")
+  end
+
+  def down
+    connection.execute("
+    do $$
+    declare
+        max_accnt_id integer;
+        curr_accnt_id integer := 0 ;
+    begin
+        execute 'select MAX(id) from accounts' into max_accnt_id;
+        IF max_accnt_id IS NOT NULL then
+            loop
+                exit when curr_accnt_id > max_accnt_id-1 ;
+                curr_accnt_id := curr_accnt_id + 1 ;
+                execute 'drop sequence IF EXISTS conv_dpid_seq_' || curr_accnt_id ;
+            end loop;
+        END IF;
+        raise notice '%', curr_accnt_id;
+    end; $$;
+    ")
+  end
```

Comment:
@sorumehta we could probably abstract this logic from raw SQL into ruby code so that it's more readable.  Let me know your thoughts. 

```suggestion
  def up
    ::Acccount.find_in_batches do |accounts_batch|
      Rails.logger.info "migrated till #{accounts_batch.first.id}\n"
      accounts_batch.each do |account|
        display_id = Conversation.where(account_id: account_id).maximum('display_id')
        display_id ||= 0 # for accounts with out conversations
        ActiveRecord::Base.connection.exec_query("create sequence IF NOT EXISTS conv_dpid_seq_#{account.id} START #{display_id + 1}")
      end
    end
  end

  def down
    ::Acccount.find_in_batches do |accounts_batch|
      Rails.logger.info "migrated till #{accounts_batch.first.id}\n"
      accounts_batch.each do |account|
        ActiveRecord::Base.connection.exec_query("drop sequence IF EXISTS conv_dpid_seq_#{account.id}")
      end
    end
  end
```

Reply:
@sojan-official  This looks great, and much more readable. Shall I click on the 'Commit suggestion' button?

Reply:
@sorumehta please go ahead. I will one more quick round of testing over that migrations.